.github/workflows/e2e-tests-contracts.yml

```diff
@@ -0,0 +1,103 @@
+name: e2e-tests-contracts
+
+on:
+  pull_request:
+    paths:
+      - 'contracts/**'
+    # branches:
+    #   - main
+    #   - "release-*"
```

Comment:
If your workflow uses caching, you should most likely build on `main`. Caches are isolated by branch, but the `main` cache is available to other branches, so if you don't build on `main`, you'll get no caching on first build of a branch.

Reply:
done